beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -2096,4 +2100,50 @@ impl<T: BeaconChainTypes> Worker<T> {
             "type" => ?message_type,
         );
     }
+
+    /// Propagate (accept) if `is_timely == true`, otherwise ignore.
+    fn propagate_if_timely(&self, is_timely: bool, message_id: MessageId, peer_id: PeerId) {
+        if is_timely {
+            // The message is still relevant, propagate.
+            self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Accept);
+        } else {
+            // The message is not relevant, ignore. It might be that this message became irrelevant
+            // during the time it took to process it, or it was received invalid.
+            self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Ignore);
+        }
+    }
+
+    /// If an attestation (agg. or unagg.) is still valid with respect to the current time (i.e.,
+    /// timely), propagate it on gossip. Otherwise, ignore it.
+    fn propagate_attestation_if_timely(
+        &self,
+        attestation: &Attestation<T::EthSpec>,
+        message_id: MessageId,
+        peer_id: PeerId,
+    ) {
+        let is_timely = attestation_verification::verify_propagation_slot_range(
+            &self.chain.slot_clock,
+            attestation,
+        )
+        .is_ok();
+
+        self.propagate_if_timely(is_timely, message_id, peer_id)
+    }
+
+    /// If a sync committee signature or sync committee contribution is still valid with respect to
+    /// the current time (i.e., timely), propagate it on gossip. Otherwise, ignore it.
+    fn propagate_sync_message_if_timely(
+        &self,
+        sync_message_slot: Slot,
+        message_id: MessageId,
+        peer_id: PeerId,
+    ) {
+        let is_timely = self
+            .chain
+            .slot_clock
+            .now()
+            .map_or(false, |current_slot| sync_message_slot == current_slot);
```

Comment:
Ahah, re-reading this made me realise I was wrong about the other PR: https://github.com/sigp/lighthouse/pull/2974#discussion_r796157356